cmd/destroy/destroy.go

```diff
@@ -435,3 +457,106 @@ func getTempKubeConfigFile(name string) string {
 	tempKubeconfigFileName := fmt.Sprintf("kubeconfig-destroy-%s-%d", name, time.Now().UnixMilli())
 	return filepath.Join(config.GetOktetoHome(), tempKubeconfigFileName)
 }
+
+func (dc *destroyCommand) runDestroyAll(ctx context.Context, opts *Options) error {
+	oktetoLog.Spinner(fmt.Sprintf("Deleting all in %s namespace", opts.Namespace))
+	oktetoLog.StartSpinner()
+	defer oktetoLog.StopSpinner()
+
+	if err := dc.oktetoClient.Namespaces().DestroyAll(ctx, opts.Namespace, opts.DestroyVolumes); err != nil {
+		return err
+	}
+
+	waitCtx, ctxCancel := context.WithCancel(ctx)
+	defer ctxCancel()
+
+	stop := make(chan os.Signal, 1)
+	defer close(stop)
+
+	signal.Notify(stop, os.Interrupt)
+	exit := make(chan error, 1)
+	defer close(exit)
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		exit <- dc.waitForNamespaceDestroyAllToComplete(waitCtx, opts.Namespace)
+	}(&wg)
+
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		err := dc.oktetoClient.Stream().DestroyAllLogs(waitCtx, opts.Namespace)
+		if err != nil {
+			oktetoLog.Warning("destroy all logs cannot be streamed due to connectivity issues")
+			oktetoLog.Infof("destroy all logs cannot be streamed due to connectivity issues: %v", err)
+		}
+	}(&wg)
+
+	select {
+	case <-stop:
+		ctxCancel()
+		oktetoLog.Infof("CTRL+C received, exit")
+		return oktetoErrors.ErrIntSig
+	case err := <-exit:
+		// wait until streaming logs have finished
+		wg.Wait()
+		return err
+	}
+}
+
+func (pc *destroyCommand) waitForNamespaceDestroyAllToComplete(ctx context.Context, namespace string) error {
+	timeout := 5 * time.Minute
+	ticker := time.NewTicker(1 * time.Second)
+	to := time.NewTicker(timeout)
+
+	c, _, err := pc.k8sClientProvider.Provide(okteto.Context().Cfg)
+	if err != nil {
+		return err
+	}
+	hasBeenDestroyingAll := false
+
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("'%s' deploy didn't finish after %s", namespace, timeout.String())
+		case <-ticker.C:
+			ns, err := c.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			status, ok := ns.Labels["space.okteto.com/status"]
+			if !ok {
+				return errors.New("namespace does not have label for status")
+			}
+
+			switch status {
+			case "Active":
+				if hasBeenDestroyingAll {
+					// when status is active again check if all resources have been correctly distroyed
+					// no configmap for the given namespace should be up and running
+					// check if there are configmaps with error state
+					cfgList, err := c.CoreV1().ConfigMaps(namespace).List(ctx, metav1.ListOptions{})
+					if err != nil {
+						return err
+					}
+					for _, cg := range cfgList.Items {
+						if cg.Data["status"] == "error" || cg.Data["status"] == "destroy-error" {
+							return fmt.Errorf("namespace destroy all failed: some resources where not destroyed")
+						}
+					}
```

Comment:
Probably checking that `len(cfgList.Items) > 0` is enough (and better). As the destroy all operation should be intended to delete everything, if a configmap is still there, it means that something failed

Reply:
> we should list all configmaps belonging to a dev environment and check if there is any in a error or destroy-error status. If there is any, the command should return an error. If there isn't any, the command should return a success

i understood that `if there is any` was regarding the configmaps errors. you meant any configmap then?

Reply:
No, at that time I was thinking just on "any configmap on that error state". But I mean, If there is any config map (regardless the state), we didn't delete all, right? 

Anyways, we can leave it as it is and change it in the future if we detect any corner case

Reply:
i think is more accurate that if the cfmap are still there, the job is not success, i will change it